pkg/steps/git_source.go

```diff
@@ -27,23 +27,19 @@ func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefiniti
 }
 
 func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
-	var refs *prowapi.Refs
-	if s.jobSpec.Refs != nil {
-		refs = s.jobSpec.Refs
-	} else if len(s.jobSpec.ExtraRefs) != 0 {
-		refs = &s.jobSpec.ExtraRefs[0]
-	} else {
-		log.Printf("Nothing to build source image from, no refs")
-		return nil
+	if refs := determineRefsWorkdir(s.jobSpec.Refs, s.jobSpec.ExtraRefs); refs != nil {
+		return handleBuild(s.buildClient, buildFromSource(s.jobSpec, "", api.PipelineImageStreamTagReferenceRoot, buildapi.BuildSource{
+			Type:       buildapi.BuildSourceGit,
+			ContextDir: s.config.ContextDir,
+			Git: &buildapi.GitBuildSource{
+				URI: fmt.Sprintf("https://github.com/%s/%s.git", refs.Org, refs.Repo),
+				Ref: refs.BaseRef,
+			},
+		}, s.config.DockerfilePath, s.resources), dry, s.artifactDir)
 	}
-	return handleBuild(s.buildClient, buildFromSource(s.jobSpec, "", api.PipelineImageStreamTagReferenceRoot, buildapi.BuildSource{
-		Type:       buildapi.BuildSourceGit,
-		ContextDir: s.config.ContextDir,
-		Git: &buildapi.GitBuildSource{
-			URI: fmt.Sprintf("https://github.com/%s/%s.git", refs.Org, refs.Repo),
-			Ref: refs.BaseRef,
-		},
-	}, s.config.DockerfilePath, s.resources), dry, s.artifactDir)
+
+	log.Printf("Nothing to build source image from, no refs")
```

Comment:
this is an error and should fail the job

Reply:
addressed